src/main/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapter.java

```diff
@@ -332,11 +332,15 @@ public Set<X509ExtensionWrapper> prepareV1Extensions(Set<Product> products, Pool
         return result;
     }
 
-    public Set<X509ExtensionWrapper> prepareV3Extensions() {
+    public Set<X509ExtensionWrapper> prepareV3Extensions(Pool pool) {
         Set<X509ExtensionWrapper> result = v3extensionUtil.getExtensions();
-        X509ExtensionWrapper typeExtension = new X509ExtensionWrapper(OIDUtil.REDHAT_OID + "." +
-            OIDUtil.TOPLEVEL_NAMESPACES.get(OIDUtil.ENTITLEMENT_TYPE_KEY), false, "Basic");
+        String entTypeOID = OIDUtil.getOid(OIDUtil.Namespace.ENTITLEMENT_TYPE);
+        String productOID = OIDUtil.getOid(OIDUtil.Namespace.PRODUCT);
+        X509ExtensionWrapper typeExtension = new X509ExtensionWrapper(entTypeOID, false, "Basic");
+        X509ExtensionWrapper productExtension = new X509ExtensionWrapper(
```

Comment:
Rather than just referencing the product's ID, this is supposed to be the namespace of the product. Currently this is not available as the work to add it has not yet been finished.